File: labs/non-native/lab3/lib/model/Flower.dart
class Flower
{
   int _id;
   String _name;
   String _type;
   String _color;
   int _number;
   double _price;
   static int _idx = 0;

   Flower(this._name, this._type, this._color, this._number, this._price){
     _id = _idx;
     _idx++;
   }

   @override
  String toString() {
    return 'Flower{name: $_name, type: $_type, color: $_color, number: $_number, price: $_price}';
  }

  int getId() {
     return _id;
  }

   String get name => _name;

  set price(double value) {
    _price = value;
  }

  set number(int value) {
    _number = value;
  }

  set color(String value) {
    _color = value;
  }

  set type(String value) {
    _type = value;
  }

  set name(String value) {
    _name = value;
  }

   String get type => _type;

   double get price => _price;

  int get number => _number;

  String get color => _color;
}

File: labs/native/app/src/main/java/com/example/lab2/model/Flower.java
package com.example.lab2.model;

import com.google.gson.annotations.SerializedName;

public class Flower {
    @SerializedName("id")
    private int id;
    @SerializedName("name")
    private String name;
    @SerializedName("type")
    private String type;
    @SerializedName("color")
    private String color;
    @SerializedName("number")
    private int number;
    @SerializedName("price")
    private double price;
    private static int index;

    public Flower(String name, String type, String color, int number, double price) {
        this.id = index;
        index++;
        this.name = name;
        this.type = type;
        this.color = color;
        this.number = number;
        this.price = price;
    }

    public Flower(int id, String name, String type, String color, int number, double price) {
        this.id = id;
        this.name = name;
        this.type = type;
        this.color = color;
        this.number = number;
        this.price = price;
    }

    public int getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public String getColor() {
        return color;
    }

    public void setColor(String color) {
        this.color = color;
    }

    public int getNumber() {
        return number;
    }

    public void setNumber(int number) {
        this.number = number;
    }

    public double getPrice() {
        return price;
    }

    public void setPrice(double price) {
        this.price = price;
    }

    @Override
    public String toString() {
        return "Flower{" +
                "name='" + name + '\'' +
                ", type='" + type + '\'' +
                ", color='" + color + '\'' +
                ", number=" + number +
                ", price=" + price +
                '}';
    }
}
